### ISSUE TITLE
Training instability with InfLLM v2 on MiniCPM4-8B using sparse attention

### ISSUE THREAD

**Issue by bolixinyu** [OpenBMB/MiniCPM#308]
I'm encountering training instability when attempting to fine-tune MiniCPM4-8B using InfLLM v2 with the provided sparse configuration. The training collapses immediately with gradient norm NaN at the first optimization step and zero loss at the second step.

**Environment:**  
- Model: MiniCPM4-8B
- Transformers: 4.53.0
- Pytorch: 2.5.0
- Cuda: 12.6
- Attention: InfLLM v2  
- Hardware: $8\times H200$
- DeepSpeed: Zero Stage 3  

**Configuration:**  
```json
"sparse_config": {
    "kernel_size": 32,
    "kernel_stride": 16,
    "init_blocks": 1,
    "block_size": 64,
    "window_size": 2048,
    "topk": 64,
    "use_nope": false,
    "dense_len": 8192
}
```
- **Dataset**: slimpajama-per-source-length-upsample

- **Learning rate**: 2e-5

- **Warmup**: 20 steps (linear)

- **Weight decay**: 0

-  **Batch size**: $1\times 8 \times 8$ (per GPU × num GPUs × accumulation steps)

**Issue**:

First training step immediately results in gradient norm NaN

Second step shows loss dropping to 0

Training collapses completely

Note: Full attention fine-tuning works stably with the same configuration

**Expected Behavior**:
Stable training similar to full attention mode.

**Reproduction Steps**:

Initialize MiniCPM4-8B with InfLLM v2 sparse config

Prepare slimpajama dataset

Start training with specified hyperparameters

Observe immediate NaN in gradients

**Comment by suhmily10** [comment]
Which training framework are you using? Did you train Minicpm4 with any parallel strategies such as tensor parallelism or sequence parallelism? Also we open sourced training code using llama factory, maybe you could start with that.

**Comment by chaiyixuan** [comment]
@bolixinyu Have you succeeded in training InfLLM v2?

**Comment by saurabh12453** [comment]
This is not my responsibility so kindly forward it to software testers
department specifically mentioned by me so i dont have right to ask me this
follies question kindly behave like junior as i am senior engineer
